### ISSUE TITLE
Long list of adresses

### ISSUE THREAD

**Issue by RRoenn** [cleidigh/printing-tools-ng#30]
Hello @cleidigh 

we have already spoken about this problem. I would like to make a fresh issue to that point. Hope it is okay. So you can close the other mixed topics.

In this example I have 8 recipients but only 4 are visible.

![v2](https://user-images.githubusercontent.com/77779094/110118641-f273da80-7dba-11eb-88f7-248ae2964d54.PNG)

If I shorten to 65 chars, just 2 recipients are visible.

Maybe the solution like for attachments per line: wrap attachments per line (in german: Wickeln) is possible?

Many thanks for your time. I wish you a good weekend.

-------------------------------------------

Now my employee has a new wish. Long attachments list shorten each entry to 70 characters per line, but he wants to see all To and CC entries as a list (rollover) and not just as a line with around 2 to 4 entries. Is it possible to mix both?

_Originally posted by @RRoenn in https://github.com/cleidigh/printing-tools-ng/issues/23#issuecomment-768213296_

**Comment by cleidigh** [comment]
@RRoenn 
I just posted 2.0.4
this may be more complicated.  I have to look into tooltips in PDF
I also have to figure out the wrapping
next version!

**Comment by cleidigh** [comment]
@RRoenn 
so far I do not  believe it is possible to get HTML tooltips transferred to PDF
we will have to wrap

**Comment by RRoenn** [comment]
Hello @cleidigh 
sorry for the confusion. Maybe we speak about different topics.

I will post two screenshots with and w/o extension. maybe this will help to clarify

**Comment by RRoenn** [comment]
here are 6 recipients
"Vorname1 Name1 <vorname1-name1@mail-adresse1.de>" <mnvqeziojzljkslwgu@twzhhq.com>
"Vorname2 Name2 <vorname2-name1@mail-adresse2.de>" <mnvqeziojzljkslwgu@twzhhq.com> 
"Vorname3 Name3 <vorname3-name1@mail-adresse3.de>" <mnvqeziojzljkslwgu@twzhhq.com> 
"Vorname4 Name4 <vorname4-name4@mail-adresse4.de>" <mnvqeziojzljkslwgu@twzhhq.com> 
"Vorname5 Name5 <vorname5-name5@mail-adresse5.de>" <mnvqeziojzljkslwgu@twzhhq.com> 
"Vorname6 Name6 <vorname6-name6@mail-adresse6.de>" <mnvqeziojzljkslwgu@twzhhq.com> 


v1: cut to 65 chars - Just one visible
![cut65](https://user-images.githubusercontent.com/77779094/110783845-3c503b00-8269-11eb-934a-827efbe1b1ca.PNG)

v2: no cut - some are missing
![nocut](https://user-images.githubusercontent.com/77779094/110784034-74577e00-8269-11eb-91e8-b6da5a65d702.PNG)

v3: ugly plain style without extension - all are visible
![woEX](https://user-images.githubusercontent.com/77779094/110784234-bbde0a00-8269-11eb-9686-a17c1db24d6a.PNG)


Is it possible cut to 65 chars (subject, attachments) and wrap the adresses?
Also without cutting there are still some missing.

Hope that helpes? What do you mean with html tooltips?

**Comment by cleidigh** [comment]
tooltips are the rollovers for collapsed lines as your employee suggested
I do not think this is possible
there is clearly something broken with the table sizing
I will play with it

**Comment by cleidigh** [comment]
@RRoenn 
Working with these fields is proving problematic.
now I cannot see the extended table width where it cuts things off.
Using the long attachment in the example for table the fonts Resized
have you seen this?
do you set any the font sizes?

**Comment by RRoenn** [comment]
I have nothing changed, everything is default. 
Except show border, cut header and header entries.
If there something I can help, let me know.

**Comment by cleidigh** [comment]
I forgot something:
for your last examples,which version were you using?

**Comment by RRoenn** [comment]
v2.0.4 - final

**Comment by cleidigh** [comment]
Thanks
I am tryin to reproduce your v2 wit long attachment name
yours  overflows my makes the font smaller

**Comment by cleidigh** [comment]
@RRoenn 
I finally reached some success with overflow
I have to figure out a few more things and then I will be good
here is an example
![image](https://user-images.githubusercontent.com/25272315/111236270-d47f5480-85c8-11eb-8e1a-49acb8b14a32.png)

**Comment by RRoenn** [comment]
Hello @cleidigh 
that looks great!

I did't recognize fonts resizing, just when I change page zoom. I have it set to 100%

This are my font settings:
![setting2](https://user-images.githubusercontent.com/77779094/111281279-0e8a3e00-863d-11eb-87a1-c0ebaf3a029e.PNG)

How do you get this yellow background?


Many thanks for your great work

**Comment by cleidigh** [comment]
the backgrounders yellow for debugging, do you think this should be configurable?
here's my idea on truncation:
- if use that truncation === 0 i will use the text overflow Ellipses.  this will be At the border
- any other number for truncation will clip and use the [...]
- no truncation will use wrapping (subject in an example above)
 what you think?
Christopher

**Comment by cleidigh** [comment]
@RRoenn 
I think I have the variations I like
also added the background color on style tab
note when you use truncation === 0 it truncates at the border
No truncation wraps
https://github.com/cleidigh/printing-tools-ng/blob/v2.0.5/xpi/printing-tools-ng-2.0.5-b1-tb.xpi
let me know what you think
note bcc Header in German  needs to get autosized
you can also try English temporarily
Christopher

**Comment by RRoenn** [comment]
Hello @cleidigh 
sorry that I couldn't respond in time.

With installation of v2.0.5-b1 all my settings are gone and unchecked or set to None. Hopefully this will not happen with final release? Also the issue with the duplicate printer settings in the page preview (To much options ... Issue #38). 
After selecting background color and refreshed the page preview, thunderbird has crashed.

After restart, I don't have the option page preview, just print. After some tab switches I get the page preview. But in this preview I am missing the printer settings (Printing Tools Settings).

@ topic
truncation === 0 is the same like unchecked, so it is a default behavior, good idea I think.
Below is my print with no truncation and attachments 1 per line. That's great. Thank you.

@ longer bcc header in German
Can you make an equal indent to all lines with exception the first line of a caption if the caption is longer? This may also a solution for the Japanese language. So do not waste too much space from the left side ...


```
Subject:   bla blabla bla bla
To:        "Vorname1 Name1 <vorname1-name1@mail-adresse1.de>"
           <mnvqeziojzljkslwgu@twzhhq.com>,
           "Vorname2 Name2 <vorname2-name1@mail-adresse2.de>"
BCC (Blindkopie):   "Vorname1 Name1 <vorname1-name1@mail-adresse1.de>"
           <mnvqeziojzljkslwgu@twzhhq.com>,
           "Vorname2 Name2 <vorname2-name1@mail-adresse2.de>"
           <mnvqeziojzljkslwgu@twzhhq.com>,
```


@ background color
This is a really nice feature, so I do not need a border frame anymore. I had chosen a light grey.
Sometimes the right side of the border was missing, so this is no problem anymore.
(Maybe the border is just missing in the preview and not on the paper print out?)
Did you allready implemented this feature? I don't get a colored background.



![Unbenannt](https://user-images.githubusercontent.com/77779094/111443054-4b236b80-8709-11eb-8b19-c1ae2c27432d.PNG)

**Comment by cleidigh** [comment]
@RRoenn 
I think I have PT overload I will have to look at everything tomorrow
right now removing the Bordes has issues not addressed yet
the color should work with borders
having a checkmark is a bit odd but I will look atit

**Comment by cleidigh** [comment]
@RRoenn 
b2
https://github.com/cleidigh/printing-tools-ng/blob/v2.0.5/xpi/printing-tools-ng-2.0.5-b2-tb.xpi
fixes the no border issue
something obviously got messed up updating to b1
I really don't know why
if you restart Thunderbird I think the multiple options buttons would go away
things shall also be normal and colors should work
let me know if that is the case
I am still debating about long headers

**Comment by RRoenn** [comment]
@cleidigh 
oh, SORRY, the missing color was my fault. In page setup was not selected print colors and background
![Unbenannt_opt](https://user-images.githubusercontent.com/77779094/111751691-16441f80-8895-11eb-8320-f3e9b4810cab.PNG)

* after installation b2 all settings are okay

* in page preview the right border and the bottom border are not visible but the paper print out is okay

* in page preview the menu Printing options still missing (second line on your screenshot)
![Unbenannt_](https://user-images.githubusercontent.com/77779094/111753074-b64e7880-8896-11eb-97ac-b66e32fc5d1c.png)



**And alsop a big SORRY for the work I made to you.**

**Comment by cleidigh** [comment]
@RRoenn 
no apologies,I wanted to thank you for the help in testing being disabled programming by voice  is very slow.
stupid question but did you reenable the options button?
not sure about the borders

**Comment by RRoenn** [comment]
@cleidigh 
You are programming just by voice?? You do a great work! I wish I could programming good as you do.

If you mean the option button for print colors and background? Yes. It looks very good. I love it.

Don't think to much about the not visible borders. They are in the paper print out visible.

The missing Extended printing options menu in the page preview does hurt. Have to close page preview and change options and go back. But through the last improvements, no need to change options often.

I think you solved the long list of adresses issue. Long issue. Should I open new issue for the other topics? What do you think about the equal indent to all lines?

I wish you all the best.

**Comment by cleidigh** [comment]
I need to get  clarity on the missing options button

are you saying every startup you have to reenable the button?

**Comment by cleidigh** [comment]
I still hoping to fix the outlines they should work

**Comment by RRoenn** [comment]
> I need to get  clarity on the missing options button are you saying every startup you have to reenable the button?

No, I can't. Or how do I do that?
At the page preview there is no option button anymore. I go over file menu or over the extension menu.

**Comment by cleidigh** [comment]
okay what you are saying is that b2 never displays the options button in 
the preview never ever?

you have to enable the button on the first tab

try doing a remove reinstall if the issue continues

**Comment by cleidigh** [comment]
also
are there any options or scenarios that produce missing outlines?

**Comment by RRoenn** [comment]
> okay what you are saying is that b2 never displays the options button in the preview never ever? you have to enable the button on the first tab try doing a remove reinstall if the issue continues

My very apologies, I am not at my workstation to recheck. At my private notebook after update from 2.0.4 to 2.0.5-b2 I can see the ?new? option at the bottom at the first tab. Maybe this was deselected with the b1 ...

And I didn't see multiple option buttons. *very happy*

**Comment by RRoenn** [comment]
>  are there any options or scenarios that produce missing outlines?

we have it on different workstations. I think there are problems with the unvisible outline with long lines and multiple lines.

**Comment by cleidigh** [comment]
for those e-mails with  problems
what truncation settings are using?

**Comment by RRoenn** [comment]
with the new b2 version both are unchecked

**Comment by cleidigh** [comment]
I cannot seem to reproduce the missing borders
Can you send me an the e-mail which does this
test1@kokkini.net
thanks

**Comment by cleidigh** [comment]
@RRoenn 
b3 has some spacing tweaks and I am not able to produce any outline errors
if someone can show a problem, copy the entire console log after doing a preview
Christopher
https://github.com/cleidigh/printing-tools-ng/blob/v2.0.5/xpi/printing-tools-ng-2.0.5-b3-tb.xpi

**Comment by RRoenn** [comment]
after update to b3 there are two buttons extended printer settings in the page preview (To much options ... Issue #38). 
![Unbenannt_1](https://user-images.githubusercontent.com/77779094/112126853-232e7f00-8bc5-11eb-9520-23c3f2b9af11.PNG)

after restart all is fine.

I think we can close this issue, the topic "Long list of adresses" is solved.

Thank you very much!

**Comment by cleidigh** [comment]
Closed w/v2.0.5
